### PR TITLE
Task to remove orphan symlinks

### LINKED ISSUE #3493: Remove old symlink when Domain is changed
I started getting random 404 at https://docs.pytest.org/en/latest/ and we found they have confifured `doc.pytest.org` as a their domain so, the new web server was failing on serving the files.

At some point, they should have configured `docs.pytest.org` under RTD and then they changed it to just `doc`, otherwise RTD should know nothing about the `docs` domain in the old web servers.

To fix this, once the Domain is changed we need to perform a cleanup of the old symlinks in the web servers. This can be done by a task that catches all the various symlink leftovers we have, and sync’s them.

* https://github.com/rtfd/readthedocs.org/blob/master/readthedocs/projects/views/private.py#L678-L701
* https://github.com/rtfd/readthedocs.org/blob/master/readthedocs/projects/forms.py#L546

> As a reference, they are mixing `doc` and `docs` in their repo: https://github.com/pytest-dev/pytest/blob/0a15edd5732578f6cc40e28770f0fc87dfa5ced9/doc/en/announce/release-3.2.0.rst

### PR DESCRIPTION
**Pull request by humitos**
This is an initial implementation of this task to remove orpahn symlinks, which should be ran periodically every 1h aprox.

In the ticket, I also suggested that the `Edit` action doesn't make sense to me and should be reconsidered.

Finally, if we consider to remove the `Edit` action, this task should be run just once and should be enough.

Closes #3493

### CODE REVIEW

File: readthedocs/projects/tasks.py [hunk 1]
```diff
@@ -778,6 +778,25 @@ def symlink_domain(project_pk, domain_pk, delete=False):
...
+    Remove orphan symlinks.
+
+    List CNAME_ROOT for Public and Private symlinks, check that all the listed
+    cname exist in the database and if doesn't exist, they are un-linked.
+    """
+    for symlink in [PublicSymlink, PrivateSymlink]:
+        for domain_path in [symlink.PROJECT_CNAME_ROOT, symlink.CNAME_ROOT]:
+            for domain in os.listdir(domain_path):
+                try:
+                    Domain.objects.get(domain=domain)
```
Comment (humitos) [inline readthedocs/projects/tasks.py]: A different approach on the implementation, making just one query to database could be like this:

```python
            valid_cnames = set(Domain.objects.all().values_list('domain', flat=True))
            orphan_cnames = set(os.listdir(domain_path)) - valid_cnames
            for cname in orphan_cnames:
                os.unlink(orphan_domain_path)
```

**Comment by ericholscher** [comment]
The issue link here is wrong. Would be good to know more about the background here. I imagine there are places we should fixing this logic in the code too.

**Comment by humitos** [comment]
@ericholscher I fixed the issue link in the description. There are some context in the issue.

> I imagine there are places we should fixing this logic in the code too.

I'm not very familiarized about how symlinks work, but this PR is _half of the solution_ that it's described in the issue.

**Comment by humitos** [comment]
@agjohnson I changed the query and did a quick test locally: all the cnames that comes with the test data (`djangokong.com` for example) were removed but `temporal.humitos.ar` (the one that I have been using for .com) was kept there.

    [09/Mar/2018 08:15:27] readthedocs.projects.tasks:795[6934]: INFO Unlinking orphan CNAME: /home/humitos/rtfd/code/readthedocs.org/private_cname_project/djangokong.com

Please, take a new a look at this and after merged I will create a new PR under `-ops` to configure the periodic task.

**Comment by agjohnson** [comment]
Any reason to not set the task up as a scheduled task in the application? I think we can keep this out of ops.

**Comment by humitos** [comment]
No strong reason, really. I though we were setup them only in the `ops` repo as I think we don't have any other scheduled task in this repository. If we have, can you point me where they are? Otherwise, can you give an example (from the web) where/how I should configure this?

**Comment by agjohnson** [comment]
We have scheduled tasks set up in our corporate repo. The ops tasks are just for tasks that only make sense in production. This is a task that would be useful for a development instance.

**Comment by humitos** [comment]
I define the task here under `CELERYBEAT` setting and opened a PR in the ops repo to extend the basic definition from here.

This should be ready to be merged.

**Comment by humitos** [comment]
I just pushed the changes requested.

I didn't find any example for celery beat and broadcast. So, I created a new task just to broadcast the one that I already had.

> The `broadcast` function could be a decorator (maybe) in the future if we need to do this multiple times.

**Comment by agjohnson** [comment]
I resolved conflicts here, going to wait on tests to merge though

**Comment by agjohnson** [comment]
Same issue with linting. Merging!

**Pull request merged**